rs/ic_os/config/src/main.rs

```diff
@@ -205,6 +274,89 @@ pub fn main() -> Result<()> {
 
             Ok(())
         }
-        None => Ok(()),
+        Some(Commands::GenerateGuestosConfig {
+            hostos_config_json_path,
+            guestos_config_json_path,
+            guestos_ipv6_address,
+        }) => {
+            let hostos_config_json_path = Path::new(&hostos_config_json_path);
+
+            let hostos_config: HostOSConfig =
+                serde_json::from_reader(File::open(hostos_config_json_path)?)?;
+
+            // TODO: We won't have to modify networking between the hostos and
+            // guestos config after completing the networking revamp (NODE-1327)
+            let mut guestos_network_settings = hostos_config.network_settings;
+            // Update the GuestOS networking if `guestos_ipv6_address` is provided
+            match &guestos_network_settings.ipv6_config {
+                Ipv6Config::Deterministic(deterministic_ipv6_config) => {
```

Comment:
What do you think about doing this in `generate_testnet_config` (if possible)?

Reply:
GenerateGuestosConfig is only used in production to convert a HostOSConfig object to a GuestOSConfig object. 

GenerateTestnetConfig is only used for testing purposes to create a GuestOSConfig object "from scratch"